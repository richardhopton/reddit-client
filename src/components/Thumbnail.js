import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";

export default class Thumbnail extends Component {
  constructor() {
    super();
    this.state = { errored: false };
  }

  handleError(event) {
    this.setState({ errored: true });
  }

  render() {
    if (this.state.errored) {
      return null;
    } else {
      const { source } = this.props;
      return (
        <div className={styles.thumbnail}>
          <img
            className={styles.thumbnailImage}
            src={source}
            onError={this.handleError.bind(this)}
          />
        </div>
      );
    }
  }
}

Thumbnail.propTypes = {
  source: PropTypes.string
};
