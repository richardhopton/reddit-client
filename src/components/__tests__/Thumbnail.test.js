import React from "react";
import Thumbnail from "../Thumbnail";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  source: "test"
};

// Check the component hasn't been accidentally adjusted by using a jest snapshot
test("renders expected snapshot", () => {
  const tree = renderer.create(<Thumbnail {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
