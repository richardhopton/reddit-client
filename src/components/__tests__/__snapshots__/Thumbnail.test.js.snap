// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expected snapshot 1`] = `
<div
  className="thumbnail"
>
  <img
    className="thumbnailImage"
    onError={[Function]}
    src="test"
  />
</div>
`;
